Formula/gismo.rb

```diff
@@ -0,0 +1,65 @@
+class Gismo < Formula
+  desc "C++ library for isogeometric analysis (IGA)"
+  homepage "https://gismo.github.io"
+  url "https://github.com/gismo/gismo/archive/refs/tags/v21.12.0.tar.gz"
+  sha256 "4001b4c49661ca8b71baf915e773341e115d154077bef218433a3c1d72ee4f0c"
+  license "MPL-2.0"
+
+  head do
+    url "https://github.com/gismo/gismo.git", branch: "stable"
+  end
+
+  depends_on "cmake" => :build
+  depends_on "openblas"
+  depends_on "suite-sparse"
+  depends_on "superlu"
+
+  on_macos do
+    depends_on "libomp"
+  end
+
+  def install
+    # Set hardware-optimized compiler flags
+    target_arch =
+      case Hardware.oldest_cpu
+      when :arm_vortex_tempest then "apple-m1"
+      else "penryn"
```

Comment:
Not sure this is correct based on https://github.com/Homebrew/brew/blob/master/Library/Homebrew/hardware.rb and https://github.com/Homebrew/brew/blob/master/Library/Homebrew/extend/os/mac/hardware.rb. Can you elaborate a bit?

Reply:
What I deduced from that file and the earlier discussions in this PR is that HomeBrew supports Apple (Arm64/x86_64) and Linux (x86_64) with Apple-M1 being the "oldest" Arm CPU to be supported and "Core2" the oldest x86_64 CPU to be supported. Due to our own CMake macro I need to map "core2" to "penryn".

Reply:
I have removed all of our compiler flags and let the code compile with whatever options are set by HomeBrew. Is this fine?